mgmt/src/models/external/overlay/vpcpeering.rs

```diff
@@ -125,4 +257,489 @@ impl VpcPeeringTable {
             .values()
             .filter(move |p| p.left.name == vpc || p.right.name == vpc)
     }
+
+    fn vpcs(&self) -> HashSet<&String> {
+        let mut vpcs = HashSet::new();
+        self.0.values().for_each(|p| {
+            vpcs.insert(&p.left.name);
+            vpcs.insert(&p.right.name);
+        });
+        vpcs
+    }
+    fn validate_peering_collisions(&self) -> ConfigResult {
+        // For each VPC, check that exposes in manifests for this VPC in the different peerings do
```

Comment:
I am not sure to understand this requirement. Why would it be bad to expose the same set of prefixes over distinct peerings?

Reply:
Between two VPCs, there should only exist one peering. You probably mean exposes here, right ?

Reply:
> Between two VPCs, there should only exist one peering.

I didn't take that into account, hence the checks.

If there's just one peering max between any pair of VPCs, then indeed I need to do fewer checks here; but I think I need to add a check elsewhere to make sure that we only have one peering at most, I don't remember seeing a check for that

Reply:
We should double-check with @Frostman and @sergeymatov , but I'd say that only one peering is needed.

I don't think we check that either atm. However, the check should be easy to do once the VPC peerings for a given VPC have been collected: there should not be two peerings with the same remote.name or remote_id. That approach requires checking the VPC's Peering objects (instead of the VpcPeering ones in the vpcpeering table), but that is somewhat more aligned with the idea of soon ceasing to use the VpcPeerings (and table), which exist because the API expresses peerings and VPCs separately, in favor of the Peering objects within VPCs.

Reply:
Sergei confirmed we only have one peering at most. I'll change my checks (see also [my comment](https://github.com/githedgehog/dataplane/pull/445#discussion_r2091577312) on the discussion above).